tests/unit/local/test_select_files.py

```diff
@@ -20,6 +20,25 @@ def test_select_files():
     assert result == expected
 
 
+def test_select_files_different_length_start_end():
+
+    files = [
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_195501-195912.nc",
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_196001-196412.nc",
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_196501-196912.nc",
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_197001-197412.nc",
+    ]
+
+    result = _select_files(files, '196201/1967')
+
+    expected = [
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_196001-196412.nc",
+        "pr_Amon_MPI-ESM1-2-HR_historical_r1i1p1f1_gn_196501-196912.nc",
+    ]
+
+    assert result == expected
```

Comment:
can you add a test case here for the reverse situation ie e.g. `result = _select_files(files, '1962/196706')` and for a completely rando combination e.g. `result = _select_files(files, '196203/196711')`?